doc/Packet-action-DONOTDROP.md

```diff
@@ -0,0 +1,63 @@
+# SAI_PACKET_ACTION_DONOTDROP packet action
+-------------------------------------------------------------------------------
+ Title       | SAI_PACKET_ACTION_DONOTDROP packet action
+-------------|-----------------------------------------------------------------
+ Authors     | Ashish Singh, Google LLC
+ Status      | In review
+ Type        | Standards track
+ Created     | 01/25/2022
+ SAI-Version | 1.8.3
+
+
+-------------------------------------------------------------------------------
+
+This spec discusses the SAI_PACKET_ACTION_DONOTDROP packet action use-cases.
+
+Goal of this packet action attribute (SAI_PACKET_ACTION_DONOTDROP) is to resolve the conflicts of ACL actions when a packet hits more than one ACLs such that when this packet action is used with an ACL, it serves as an action that prevents the possible drop from another ACL that is in a lower priority table.
+
+This new packet action attribute is in line with the current SAI architecture - actions in higher priority groups take precedence. If an ACL with DROP action is in a higher priority group than this ACL then, DROP action is expected to be honored. Few cases are explained with examples in next section.
+
```

Comment:
What is the behavior if  the highest priority entry in ACL stage 1 (say ingress) marks as DO_NOT_DROP, and another entry in ACL Stage 2 (say egress) marks the packet as drop? 

In order to avoid ambiguities, we should spell out that DO NOT DROP overrides the DROP action set by:

1) ACL tables with lower priority in this ACL Table Group, AND

2) ACL tables in preceding stages


In other words, it would not override the drop action set in subsequent ACL Table stages.

Reply:
As per my understanding, ACL action resolution is done per-stage and an ingress stage DONOTDROP action should not have have impact on VFP or EFP actions.

Reply:
Updated the doc with comment on DONOTDROP action in one stage will not override the drop action is subsequent ACL stages. Thank you.